PURL-SPECIFICATION.rst

```diff
@@ -247,8 +247,9 @@ Use these rules for percent-encoding and decoding ``purl`` components:
 - the '#', '?', '@' and ':' characters must NOT be encoded when used as
   separators. They may need to be encoded elsewhere
 
-- the ':' ``scheme`` and ``type`` separator does not need to and must NOT be encoded.
-  It is unambiguous unencoded everywhere
+- The colon ':' separator between ``scheme`` and ``type`` MUST NOT be encoded.
```

Comment:
I guess that following @JimFuller-RedHat suggestion in today's triage call https://github.com/package-url/purl-spec/discussions/360 ... this could reworded as:
```suggestion
- The colon ':' separator between ``scheme`` and ``type`` MUST be used as-is, unencoded.
```

@JimFuller-RedHat ?

Reply:
Hehe, I was (badly) trying say that specs typically define grammar, schema and behavior (ex. processing rules) ... in this case I think how you had it was fine. It is easier to define a 'thing' precisely which has the side effect of defining all the things it cannot be ;)

From grammar pov we could define (in EBNF) as a literal colon ... any parser built with such an EBNF would bork on encoded input ... we might suggest defining pre processing rules to normalise odd input into canonical purl. or define a looser grammar allowing encoding. 

I would propose:

_"The``scheme`` and ``type`` MUST be separated by a colon ':' "_

Reply:
@JimFuller-RedHat 
> I would propose:
> 
> _"The`scheme` and `type` MUST be separated by a colon ':' "_

clean and neat.

Yet someone would likely immediately bet a nickel that someone else will open an issue asking whether the colon can or must or must not be encoded. ;)

Here is the way out:
We can keep these negative assertions as comments in a new FAQ document. I think that we can create a leaner spec this way.

I think it makes sense for the spec to avoid specifying all the things that it does not support as you rightly suggest, because there is an infinity of these. And we can push "negative" support questions in that FAQ.

So here:

- In the spec:
  - `The scheme and type MUST be separated by a colon ':'`

- In the FAQ:
  - Q: Is the colon between scheme and type encoded? Or can it be encoded? If yes, how?
  - A: The spec has no mention of encoding there, so the colon should be used as-is, never encoded and never requiring any decoding since this is not encoded. It is a parsing error if the colon does not come directly after `pkg`; Lenient parsing tools are welcome to recover from this error to help process and sanitize damaged purls, but that's not a requirement, and not part of the spec.

That approach can become the standard. The spec will be leaner and crisper and a the FAQ comprehensive to address the questions that _will_ come up for sure.

Reply:
cool ! fantastic suggestion - brevity and conciseness enables implementators to focus ... a good example is the json spec - https://ecma-international.org/wp-content/uploads/ECMA-404_2nd_edition_december_2017.pdf , which only contains normative text ;)